frontend/apps/reader/modules/readersearch.lua

```diff
@@ -287,12 +287,12 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
                 self.wait_button.movable:setMovedOffset(self.search_dialog.movable:getMovedOffset())
                 UIManager:show(self.wait_button)
                 UIManager:tickAfterNext(function()
-                    do_search(func, pattern, param, regex, case_insensitive)()
+                    do_search(func, pattern, param)()
```

Comment:
This smells like the wrong solution to an accidental regression.

Reply:
Ought to be safe, everything in the same function, so these are closure'ed earlier, looks like.

Haven't looked at the history, though.

Reply:
Possibly clearer to just keep the larger signature, and populate it instead, though?

Reply:
I suppose it's more of a comment on the commit message than on the change per se. As written it sounds a lot like "apply suggestion from dumb linter that doesn't understand context," which I find unhelpful at best. The question is if it's unnecessary, not if the number of arguments is mismatched. Simply removing some arguments to shut up a linter smells like papering over a deeper problem.

You might say it's similar in that I'd still have to check if it had said something like "unnecessary," but then it's double checking a human, not a dumb linter.

Reply:
> Possibly clearer to just keep the larger signature, and populate it instead, though?

Also that.

Reply:
Looks indeed like these 2 args are not used.
Pinging @zwim from #7883 - to check if this is alright and that it doesn't hide some other problem (a bit head scratching these closures & callbacks...)

Reply:
My understanding of the code was that `do_search()` was now wrapped in another container where the context (regex, ci) was defined. So the old parameters of `do_search()` were no longer needed. Of course, this is not visible in the patch, and the source code of the search is quite long, so I could have misunderstood.